advanced_source/torch_script_parallelism.rst

```diff
@@ -0,0 +1,276 @@
+Dynamic Parallelism in TorchScript
+==================================
+
+In this tutorial, we introduce the syntax for doing *dynamic inter-op parallelism*
+in TorchScript. This parallelism has the following properties:
+
+* dynamic - The number of parallel tasks created and their workload can depend on the control flow of the program.
+* inter-op - The parallelism is concerned with running TorchScript program fragments in parallel. This is distinct from *intra-op parallelism*, which is concerned with splitting up individual operators and running subsets of the operator's work in parallel.
+Basic Syntax
+------------
+
+The two important APIs for dynamic parallelism are:
+
+* ``torch.jit.fork(fn : Callable, *args, **kwargs) -> torch.jit.Future[T]``
+* ``torch.jit.wait(fut : torch.jit.Future[T]) -> T``
+
+A good way to demonstrate how these work is by way of an example:
+
+.. code-block:: python
+
+    import torch
+
+    def foo(x):
+        return torch.neg(x)
+
+    @torch.jit.script
+    def example(x):
+        # Call `foo` normally
+        x_normal = foo(x)
+
+        # Call `foo` using parallelism:
+        # First, we "fork" off a task. This task will run `foo` with argument `x`
+        future = torch.jit.fork(foo, x)
```

Comment:
maybe makes sense to put the fork call before the normal call to emphasize it executes asynchronously